cra-kit: correct Art. 14 reporting mechanics (SRP/CSIRT/EUVD) and remove deleted ROADMAP references (#603)
wolfSSL/wolfssl-examples: cra-kit/wolfssl-inc-auditor-packet/vulnerability-handling-process.md

# Vulnerability handling process — wolfSSL Inc.

> **Status:** ✅ wolfSSL Inc. commits to complying with its CRA Art. 13/14 obligations

Reviewer: Not a review call, so I am not suggesting a change: flipping this to a committed Art. 13/14 statement, along with the on-call rewrite below, is a public compliance commitment rather than an accuracy fix. It wants its own PR and its own sign-off. Everything else here I am happy with once the citations are fixed.

Worth knowing for that PR: Art. 14 applies from 11 Sep 2026, and Art. 69(3) derogates from the usual grandfather rule so it reaches products placed on the market before 11 Dec 2027. The installed base is in scope, not just new releases.

            ▼                        ▼
   ┌────────────────┐       ┌──────────────────┐
   │ Triage (72h):  │       │ Active exploit?  │
   │ severity, CVSS,│ ────▶ │ Yes ─▶ SRP 24h   │

>    │ scope, fix plan│       │ No  ─▶ standard   │

Reviewer: The box is one column wide here. Border is 18 dashes; this content line is 19. It was 17 before this PR, so the edit flipped the error rather than fixing it.

```suggestion
   │ scope, fix plan│       │ No  ─▶ standard  │
```

   └────────────────┘
```

## How an Art. 14 report is filed (SRP → CSIRT + ENISA → EUVD)

Reports are **not** sent to ENISA directly. Under Art. 14/16 a manufacturer files
through the **Single Reporting Platform (SRP)** operated by ENISA, using the
electronic notification end-point of the **CSIRT designated as coordinator**; ENISA
receives the notification **simultaneously**. The coordinator CSIRT then disseminates
it to the CSIRTs of other affected Member States and to market-surveillance

> authorities.

Reviewer: Art. 14(1) requires notification "simultaneously to the CSIRT designated as coordinator ... **and to ENISA**", and 14(7) has the submission "simultaneously accessible to ENISA". ENISA is a co-addressee by statute rather than a copy recipient, so "not sent to ENISA directly" swings a bit too far. The old text was imprecise about the mechanism, not wrong about ENISA.

```suggestion
A manufacturer files **once**, through the **Single Reporting Platform (SRP)**
operated by ENISA. Art. 14(1) requires notification **simultaneously to the CSIRT
designated as coordinator and to ENISA**, and Art. 14(7) directs that submission to
the coordinator CSIRT's electronic notification end-point, "simultaneously accessible
to ENISA". The coordinator CSIRT then disseminates it to the CSIRTs of other affected
Member States and to market-surveillance authorities.
```

- **Which CSIRT?** Art. 14(7) ties the coordinator CSIRT to the manufacturer's EU
  main establishment. wolfSSL Inc. is US-established with **no EU main establishment**,
  so the coordinator is determined by the Member State of its **EU Authorised
  Representative** (Art. 18) — see [`eu-authorised-representative.md`](eu-authorised-representative.md).

>   Resolving the AR therefore also fixes the reporting end-point.

Reviewer: Art. 14(7) sets a four-step cascade where there is no EU main establishment, not a single AR rule: (a) the AR acting for the highest number of that manufacturer's products, then (b) importer, (c) distributor, (d) Member State with the most users.

Also relevant: we have decided **not** to appoint an Art. 18 authorised representative. Art. 18(1) says a manufacturer *may* appoint one, and every duty in the Art. 18(3) mandate is one we already carry under Art. 13(13). So the endpoint resolves through the rest of the cascade.

```suggestion
- **Which CSIRT?** Art. 14(7) ties the coordinator CSIRT to the manufacturer's EU
  main establishment. Where there is none, the same paragraph sets an ordered
  cascade: (a) the Member State of the authorised representative acting for the
  highest number of that manufacturer's products, then (b) the importer, (c) the
  distributor, and (d) the Member State in which the highest number of users are
  located. wolfSSL Inc. is US-established with **no EU main establishment**, so its
  coordinator is fixed by that cascade.
```

- **Platform availability.** The SRP goes live **11 Sep 2026**, the date Art. 14
  reporting begins. Manufacturers (or their AR) register via an **EU Login** account;

>   the coordinator CSIRT validates the representative's authority to file.

Reviewer: Two things worth correcting here, and the first could cost someone the 24-hour clock.

ENISA's SRP guidance says the coordinator CSIRT's validation runs *in parallel with* reporting and "is not a prerequisite for fulfilling the CRA reporting obligation". As written this reads as a gate on filing.

Second, ENISA's "AR" in that guidance means **Assigned Representative**, a platform user role with primary/secondary seats. That is not the Art. 18 **authorised representative** this doc links to elsewhere. Easy trap.

Ref: https://www.enisa.europa.eu/topics/product-security/single-reporting-platform-srp/cra-srp-guidance-ar-user-registration

```suggestion
- **Platform availability.** The SRP goes live **11 Sep 2026**, the date Art. 14
  reporting begins. Filers register through an **EU Login** account. ENISA's guidance
  states that the coordinator CSIRT's validation of a platform user's authority runs
  *in parallel with* reporting and is **not a prerequisite for fulfilling the CRA
  reporting obligation**, so validation status never blocks a submission. Note that
  ENISA's "Assigned Representative" is a platform user role, **not** the Art. 18
  authorised representative.
```

- **Where fixed vulnerabilities are published.** Once a report is filed and a fix is
  available, ENISA publishes the entry to the **European Vulnerability Database

>   (EUVD)** (Art. 16(2)) — making the disclosure timeline publicly verifiable.

Reviewer: EUVD publication is **Art. 17(5)**, not 16(2). Art. 16(2) is CSIRT-to-CSIRT dissemination. 17(5) also carries a qualifier worth keeping: ENISA adds the entry *"in agreement with the manufacturer"*. The EUVD itself is established under NIS2 Art. 12(2), not by the CRA.

```suggestion
- **Where fixed vulnerabilities are published.** Once a corrective or mitigating
  measure is available, ENISA adds the notified vulnerability to the **European
  Vulnerability Database (EUVD)**, established under NIS2 Art. 12(2), **in agreement
  with the manufacturer** (Art. 17(5)) — making the disclosure timeline publicly
  verifiable.
```

| CVE Numbering Authority | wolfSSL is a [CNA](https://www.cve.org/PartnerInformation/ListofPartners) | wolfSSL assigns CVE IDs within the wolfSSL libraries scope |

## Service-level targets

| Stage | Target | Notes |
|-------|--------|-------|
| Acknowledgement of receipt | **24 hours** | From any channel listed in `security.txt`. |
| Initial triage (severity, validity, fix plan) | **72 hours** | |
| Early-warning notification (SRP → CSIRT + ENISA) | **24 hours from awareness of active exploitation** (Art. 14(2)(a)) | Hard regulatory deadline — not negotiable. |
| Follow-up notification (via SRP) | **72 hours from awareness** (Art. 14(2)(b)) | Hard regulatory deadline. |
| Final report (via SRP) | **14 days after a corrective or mitigating measure is available** (Art. 14(2)(c)) | Hard regulatory deadline. Clock runs from fix-availability, **not** from awareness or CVE publication. |
| Coordinated public disclosure | Typically 90 days from triage; case-by-case | Negotiable with reporter. |

wolfSSL Inc. commits to meeting these Art. 14 deadlines for wolfSSL libraries it
places on the EU market. The acknowledgement and triage targets above are also

> reflected in the public CVD policy at `/.well-known/vulnerability-disclosure-policy.txt`.

Reviewer: This is the one blocker for me. The published policy has no 24h acknowledgement and no 72h triage target: it says "We will acknowledge receipt **without undue delay**", the word "triage" does not appear, and there is no hour count anywhere on the page. An auditor follows that link and the packet takes the hit.

Either the numbers go onto the published policy first, or the clause comes out.

```suggestion
These targets are not yet publicly committed in the CVD policy. Once the
leadership decision is taken, the CVD policy at `/.well-known/vulnerability-disclosure-policy.txt`
will be updated to include them.
```

**Severe incidents (Art. 14(3)).** The same 24h early-warning / 72h notification
cadence also applies to **severe incidents** affecting the security of the product,
with a **final report within one month** of the 72-hour notification (rather than the

> 14-day-after-fix deadline that applies to vulnerabilities).

Reviewer: Art. 14(3) is the duty to notify severe incidents; the deadlines are in Art. 14(4), with the one-month final report at 14(4)(c). Your 14(2)(a)-(c) cites on the rows above are correct as-is, since that is the actively-exploited-vulnerability track.

```suggestion
**Severe incidents (Art. 14(3)–(4)).** Art. 14(3) extends the same 24h early-warning
/ 72h notification cadence to **severe incidents** affecting the security of the
product. Those deadlines sit in Art. 14(4): (a) 24 hours, (b) 72 hours, and (c) a
**final report within one month** of the 72-hour notification, rather than the
14-day-after-fix deadline that applies to vulnerabilities under Art. 14(2)(c).
```
